MyListLibrary/LinkedList.cs

```diff
@@ -0,0 +1,406 @@
+                {
+                    throw new IndexOutOfRangeException();
+                }
+
+                GetNode(index).Value = value;
+            }
+        }
+
+        private Node<T> GetNode(int index)
+        {
+            Node<T> temp = root;
+            for (int i = 0; i < index; i++)
+            {
+                if (temp.Next == null)
+                {
+                    throw new IndexOutOfRangeException();
+                }
+
+                temp = temp.Next;
+            }
+
+            return temp;
+        }
+
+        public void AddByIndex(int index, T value)
+        {
+            if (index < 0 || index > Count)
+            {
+                throw new IndexOutOfRangeException();
+            }
+
+            Node<T> newNode = new(value);
+
+            if (index != Count)
+            {
+                newNode.Next = GetNode(index);
+            }
+
+            if (index != 0)
+            {
+                GetNode(index - 1).Next = newNode;
+            }
+            else
+            {
+                root = newNode;
+            }
+
+            ++Count;
+        }
+
+        public void AddElementsByIndex(int index, IMyList<T> list)
+        {
+            if (index > Count)
+            {
+                throw new IndexOutOfRangeException();
+            }
+
+            LinkedList<T> newList;
+
+            if (!(list is LinkedList<T>))
+            {
+                newList = ToLinkedList(list);
+            }
+            else
+            {
+                newList = (LinkedList<T>)list;
+            }
+
+            if (root == null)
+            {
+                root = newList.root;
+            }
+
+            Node<T> curr = GetNode(index);
+
+            if (index != 0)
+            {
+                GetNode(index - 1).Next = newList.GetNode(0);
+            }
+
+            if (index != Count)
+            {
+                newList.GetNode(newList.Count - 1).Next = curr;
+            }
+
+            Count += list.Count;
+        }
+
+        public void AddElementsFirst(IMyList<T> list)
+        {
+            AddElementsByIndex(0, list);
+        }
+
+        public void AddElementsLast(IMyList<T> list)
+        {
+            AddElementsByIndex(Count, list);
+        }
+
+        public void AddFirst(T value)
+        {
+            AddByIndex(0, value);
+        }
+
+        public void AddLast(T value)
+        {
+            AddByIndex(Count, value);
+        }
+
+        public int GetIndexFirstElementByValue(T value)
+        {
+            Node<T> current = root;
+            int index = -1;
+
+            for (int i = 0; i < Count; i++)
+            {
+                if (current.Value.Equals(value))
+                {
+                    index = i;
+                }
+
+                current = current.Next;
+            }
+
+            return index;
+        }
+
+        public int GetIndexMaxElement()
+        {
+            Node<T> current = root;
```

Comment:
1. Create private method for return node with MaxElement and index.
2. Reuse it in max and maxI methods

Reply:
good point, fixed